### PR TITLE
Embed subscriber service

### PR DESCRIPTION
**Pull request by Jason-ZW**
ROS upgrade/subscriber service

### CODE REVIEW

File: os-config.tpl.yml [hunk 1]
```diff
@@ -381,6 +381,25 @@ rancher:
...
+      privileged: true
+      pid: host
+      ipc: host
+      net: host
+      uts: host
+      volumes_from:
+      - command-volumes
+      - system-volumes
+      volumes:
+      - /sys:/sys
```
Comment (niusmallnan) [inline os-config.tpl.yml]: better to use `/sys:/host/sys`.

File: images/01-base/Dockerfile [hunk 2]
```diff
@@ -39,7 +39,8 @@ RUN rm /sbin/poweroff /sbin/reboot /sbin/halt && \
     ln -s /usr/share/dhcpcd/hooks/10-wpa_supplicant /lib/dhcpcd/dhcpcd-hooks/ && \
     rm -f /usr/share/bash-completion/completions/* && \
     sed -i 1,10d /etc/rsyslog.conf && \
-    echo "*.*                /var/log/syslog" >> /etc/rsyslog.conf
+    echo "*.*                /var/log/syslog" >> /etc/rsyslog.conf && \
+    chmod +x /usr/bin/subscriber
```
Comment (niusmallnan) [inline images/01-base/Dockerfile]: better to use `os-subscriber`

File: os-config.tpl.yml [hunk 3]
```diff
@@ -399,6 +418,7 @@ rancher:
   upgrade:
     url: {{.OS_RELEASES_YML}}/releases{{.SUFFIX}}.yml
     image: {{.OS_REPO}}/os
+    policy: "download"
```
Comment (niusmallnan) [inline os-config.tpl.yml]: Can consider removing double quotes.

File: images/01-base/Dockerfile (hunk 2 above)
Comment (niusmallnan) [reply images/01-base/Dockerfile]: And we don't need `chmod` if we pre-set permission for this file.

File: os-config.tpl.yml (hunk 1 above)
Comment (Jason-ZW) [reply os-config.tpl.yml]: Already modified

File: images/01-base/Dockerfile (hunk 2 above)
Comment (Jason-ZW) [reply images/01-base/Dockerfile]: Already modified

File: os-config.tpl.yml (hunk 3 above)
Comment (Jason-ZW) [reply os-config.tpl.yml]: Already modified

**Pull request merged**